crates/yttrium/src/pay/mod.rs

```diff
@@ -22,30 +21,367 @@ impl<T: std::fmt::Debug> From<progenitor_client::Error<T>> for PayError {
+            amount: r.amount.into(),
+            expires_at: r.expires_at,
+            options: r.options.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+// ==================== Client ====================
+
 #[cfg_attr(feature = "uniffi", derive(uniffi::Object))]
 pub struct WalletConnectPay {
     client: Client,
+    config: SdkConfig,
 }
 
 #[cfg_attr(feature = "uniffi", uniffi::export(async_runtime = "tokio"))]
 impl WalletConnectPay {
     #[cfg_attr(feature = "uniffi", uniffi::constructor)]
-    pub fn new(base_url: &str) -> Self {
-        let client = Client::new(base_url);
-        Self { client }
+    pub fn new(base_url: String, config: SdkConfig) -> Self {
+        let client = Client::new(&base_url);
+        Self { client, config }
     }
 
+    /// Simple test method to verify bindings are loaded correctly
+    pub fn sum(&self, a: i64, b: i64) -> i64 {
+        a + b
+    }
+
+    // ==================== Merchant API ====================
+
+    /// Create a new payment
     pub async fn create_payment(
         &self,
+        merchant_id: String,
         reference_id: String,
         amount_unit: String,
         amount_value: String,
     ) -> Result<CreatePaymentResponse, PayError> {
-        let body = CreatePayment {
+        let body = types::CreatePayment {
             reference_id,
-            amount: Amount { unit: amount_unit, value: amount_value },
+            amount: types::Amount { unit: amount_unit, value: amount_value },
         };
-        let response = self.client.create_payment().body(body).send().await?;
-        Ok(response.into_inner())
+        let response = with_retry(|| async {
+            self.client
+                .create_payment_handler()
+                .api_key(&self.config.api_key)
+                .merchant_id(&merchant_id)
+                .sdk_name(&self.config.sdk_name)
+                .sdk_version(&self.config.sdk_version)
+                .sdk_platform(&self.config.sdk_platform)
+                .body(body.clone())
+                .send()
+                .await
+        })
+        .await?;
+        Ok(response.into_inner().into())
+    }
+
+    /// Get payment status (for polling)
+    pub async fn get_payment_status(
+        &self,
+        merchant_id: String,
+        payment_id: String,
+    ) -> Result<GetPaymentStatusResponse, PayError> {
+        let response = with_retry(|| async {
+            self.client
+                .get_payment_status_handler()
+                .api_key(&self.config.api_key)
+                .merchant_id(&merchant_id)
+                .sdk_name(&self.config.sdk_name)
+                .sdk_version(&self.config.sdk_version)
+                .sdk_platform(&self.config.sdk_platform)
+                .id(&payment_id)
+                .send()
+                .await
+        })
+        .await?;
+        Ok(response.into_inner().into())
+    }
+
+    /// Poll for payment status until it reaches a final state or timeout
```

Comment:
Wallet doesn't need polling. The confirmPayment will be blocking and will return payment success or error, check:

https://www.notion.so/walletconnect/WalletConnect-Pay-WX-Technical-Design-2bd3a661771e8026bd97f52c78b688fa?source=copy_link#2bd3a661771e80dcbc62d4ea11a598b2
https://www.notion.so/walletconnect/WalletConnect-Pay-WX-Technical-Design-2bd3a661771e8026bd97f52c78b688fa?source=copy_link#2bd3a661771e80479a88d9a8dd64dded